docs/docs/learn/programming/mcp.md

```diff
@@ -0,0 +1,187 @@
+---
+sidebar_position: 3
+---
+
+# Model Context Protocol (MCP)
+
+The [Model Context Protocol (MCP)](https://modelcontextprotocol.io/) is an open protocol that standardizes how applications provide context to language models. DSPy supports MCP, allowing you to use tools from any MCP server with DSPy agents.
+
+## Installation
+
+Install DSPy with MCP support:
+
+```bash
+pip install -U "dspy[mcp]"
+```
+
+## Overview
+
+MCP enables you to:
+
+- **Use standardized tools** - Connect to any MCP-compatible server
+- **Share tools across stacks** - Use the same tools across different frameworks
+- **Simplify integration** - Convert MCP tools to DSPy tools with one line
+
+As long as you have an `mcp.ClientSession`, you can use MCP tools with DSPy's `ReAct` module or any other tool-based workflow.
```

Comment:
I think users will be confused about "As long as you have an `mcp.ClientSession`", because it's not a well-known thing for many people.

We should maybe add a short paragraph to explain how DSPy manages the connections to the MCP server - we utilize the native MCP session for all connection types, and don't provide extra features for connection handling